Sprint_6

FaizullinMarsel opened the pull request:
Добавлены три тестовых класса CatTest, LionTest, FelineTest 
В классе Lion применена инъекция зависимостей
В интерфейсе Predator добавлен метод getKittens

FaizullinMarsel committed:
UI Тестирование
Добавлены три тестовых класса CatTest, LionTest, FelineTest
В классе Cat применена инъекция зависимостей
В интерфейсе Predator добавлен метод getKittens

FaizullinMarsel committed:
add jacoco report

FaizullinMarsel committed:
Отформатирован код при помощи горячих клавиш Ctrl+Alt+L
Удалены лишние комментарии
Переименованы переменные для Mock

FaizullinMarsel committed:
Удалена лишния зависимость junit в Pom файле

egorvas commented on src/main/java/com/example/Lion.java:
```diff
@@ -12,21 +12,26 @@ public Lion(String sex) throws Exception {
 
+    Predator predator;
+
+    public Lion(Feline feline) {
```
У тебя два конструктора для класса Lion (Feline feline) и public Lion(String sex) throws Exception.
Если создавать экземпляр класса Lion, используя конструктор Lion (Feline feline), то для экземпляра класса hasMane будет всегда false -- это плохо, так делать нельзя.
Если создавать экземпляр класса Lion, используя конструктор Lion(String sex), то для экземпляра класса не будет определен feline -- это плохо, так делать нельзя.
Тебе нужно реализовать один конструктор для класса Lion, в котором будет определен и feline и hasMane.
После этого нужно подправить тесты для этого класса.

FaizullinMarsel committed:
1. Исправлен конструктор в классе Lion
2. Исправлены наименования всех тестовых методов
3. Добавлен новый тест проверяющий исключение в классе LionTest
4. Откорректированы тексты ошибок во всех тестах
5. В классе LionTest удалены лишние тестовые данные для проверки
6. В классах shouldCheckKittens и shouldCheckAmountKittens_WithConstructor удалены лишние переменные типа int
